File: Wipro_Capstone_project/FRONTEND/CapstoneProjectAngular/src/app/interface/order.ts
// order.model.ts
export interface Order {
  orderId: number;
  userId: number;
  productQuantityMap: { [productId: number]: number }; // Map<Integer, Integer>
  totalAmount: number;
  orderStatus: string;
  orderDate: string; // can use string for LocalDateTime
}

export interface OrderDTO {
  userId: number;
  productQuantityMap: { [productId: number]: number };
  totalAmount: number;
  status?: string; // optional, default PENDING
}

File: Wipro_Capstone_project/BACKEND/order-ms/order-ms/src/main/java/com/backend/order_ms/entity/Order.java
package com.backend.order_ms.entity;

import java.time.LocalDateTime;
import java.util.Map;

import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "orders")
public class Order {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int orderId;
    
    private int userId;
    
    @ElementCollection
    private Map<Integer, Integer> productQuantityMap; // productId -> quantity
    
    private double totalAmount;
    
    private String orderStatus;
    
    private LocalDateTime orderDate;
}
